examples/BasicOperations/GetResponsiveSearchAds.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            // Get the GoogleAdsService.
+            GoogleAdsServiceClient googleAdsService = client.GetService(
+                Services.V3.GoogleAdsService);
+
+            string searchQuery =
+                $@"SELECT
+                 ad_group.id,
+                 ad_group_ad.ad.id,
+                 ad_group_ad.ad.responsive_search_ad.headlines,
+                 ad_group_ad.ad.responsive_search_ad.descriptions,
+                 ad_group_ad.status
+            FROM ad_group_ad
+            WHERE
+                ad_group_ad.ad.type = RESPONSIVE_SEARCH_AD
+                AND ad_group_ad.status != 'REMOVED'";
+
+            if (adGroupId != null)
+            {
+                searchQuery += $" AND ad_group.id = {adGroupId}";
+            }
+
+            // Create a request that will retrieve all ads using pages of the specified page size.
+            SearchGoogleAdsRequest request = new SearchGoogleAdsRequest()
+            {
+                CustomerId = customerId.ToString(),
+                PageSize = PAGE_SIZE,
+                Query = searchQuery
+            };
+
+            try
+            {
+                // Issue the search request.
+                PagedEnumerable<SearchGoogleAdsResponse, GoogleAdsRow> searchPagedResponse =
+                    googleAdsService.Search(request);
+
+                // Iterate over all rows in all pages and prints the requested field values for
+                // the ad in each row.
+                foreach (GoogleAdsRow googleAdsRow in searchPagedResponse)
+                {
+                    Ad ad = googleAdsRow.AdGroupAd.Ad;
+
+                    Console.WriteLine($"Responsive search ad with resource name '{ad.ResourceName}'," +
+                        $"status '{googleAdsRow.AdGroupAd.Status}' was found.");
+                    // Prints the ad text asset detail.
+                    ResponsiveSearchAdInfo responsiveSearchAdInfo = ad.ResponsiveSearchAd;
+                    Console.WriteLine("Headlines:{0},\nDescriptions:{1}",
+                        AdTextAssetsToStrings(responsiveSearchAdInfo.Headlines),
+                        AdTextAssetsToStrings(responsiveSearchAdInfo.Descriptions));
+                }
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+                throw;
+            }
+        }
+
+        /// <summary>Converts the the text assets to strings.</summary>
+        /// <param name="adTextAssets">The ad text assets.</param>
+        /// <returns>The string representation of the provided list of AdTextAsset
+        /// objects.</returns>
+        private static string AdTextAssetsToStrings(RepeatedField<AdTextAsset> adTextAssets)
```

Comment:
The name of the function should be "AdTextAssetsToString~~s~~": it only returns one string

Reply:
Done.